sw/device/lib/base/print.h

```diff
@@ -0,0 +1,135 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef SW_DEVICE_LIB_BASE_PRINT_H_
+#define SW_DEVICE_LIB_BASE_PRINT_H_
+
+#include <stdarg.h>
+#include <stddef.h>
+
+/**
+ * This header provides libc-like printing facilities, which is agnostic of the
+ * underlying hardware printing mechanism.
+ */
+
+/**
+ * A buffer_sink_t represents a place to write bytes to, implemented as a
+ * C-style "closure".
+ *
+ * It consists of a generic data pointer, which can hold instance-specific
+ * information, and a sink function, which takes the data pointer, a buffer, and
+ * that buffer's length.
+ *
+ * The sink function should return the number of bytes actually written.
+ */
+typedef struct buffer_sink {
+  void *data;
+  size_t (*sink)(void *data, const char *buf, size_t len);
+} buffer_sink_t;
+
+/**
+ * Prints out a message to stdout, formatted according to the format string
+ * |format|.
+ *
+ * The definition of "stdout" is not provided by this library; rather, it must
+ * be initialized using |base_set_stdout()|.
+ *
+ * This function supports a subset of the format specifiers provided by standard
+ * C |printf|. Those are, namely:
+ * - %%, which prints a percent sign.
+ * - %c, which prints a single ASCII character.
+ * - %s, which prints a NUL-terminated string.
```

Comment:
Please add a note indicating string is not supported when writing tests intended for DV.

Reply:
Done. I've also added a similar comment for %z, which I don't think will work either.